bee-message/src/output/foundry.rs

```diff
@@ -442,9 +448,12 @@ impl StateTransitionVerifier for FoundryOutput {
                     return Err(StateTransitionError::InconsistentNativeTokensMeltBurn);
```

Comment:
I can only comment on this line because the previous 3 are not part of this PR, but I just want to make sure the above if-condition is correct: Currently only if both inqualities are true, we return this state transition error, but what if only one of those is true? Isn't that inconsistent already?

Reply:
Yes this is correct. It means that you are not allowed to mint while melting but you are allowed to mint while burning. It implements the following rules from https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#token-scheme-semantic-validation-rules `When Current State(Melted Tokens) != Next State(Melted Tokens), it must be true that Current State(Minted Tokens) = Next State(Minted Tokens)`

Reply:
Oh no, I looked at the wrong line.

Reply:
All good then ?

Reply:
Honestly, I still have a hard time why the melting is not mirroring the minting transition :thinking: I feel like I am looking at a bug here, but I am probably having a misunderstanding.

Reply:
ah, okay, didn't see you message. All good then :+1:

Reply:
Because `Ordering::Greater` is about melting + burning, not only melting. Going for lunch now, can answer questions later.

Reply:
me too, need some lunch